src/main/java/org/folio/service/assignedusers/AssignedUsersServiceImpl.java

```diff
@@ -40,12 +48,53 @@
   private Converter<DbAssignedUser, AssignedUser> fromDbConverter;
   @Autowired
   private Converter<AssignedUser, User> userConverter;
+  @Autowired
+  private UsersLookUpService usersLookUpService;
 
+  @SneakyThrows
   @Override
   public CompletableFuture<AssignedUserCollection> findByCredentialsId(String credentialsId,
                                                                        Map<String, String> okapiHeaders) {
-    return assignedUserRepository.findByCredentialsId(toUUID(credentialsId), tenantId(okapiHeaders))
+    var dbAssignedUsers = assignedUserRepository.findByCredentialsId(toUUID(credentialsId), tenantId(okapiHeaders));
+
+    var ids = dbAssignedUsers
+      .thenApply(dbAssignedUsers1 -> dbAssignedUsers1.stream()
+        .map(DbAssignedUser::getId)
+        .collect(Collectors.toList()));
+
+    var users = new ArrayList<CompletableFuture<User>>();
+
+    ids.thenApply(idList -> idList.stream().map(id -> {
+      okapiHeaders.put(XOkapiHeaders.USER_ID, String.valueOf(id));
+      return users.add(usersLookUpService.lookUpUser(new OkapiParams(okapiHeaders)));
+    }));
```

Comment:
You should request a collection of users in one request. For this purpose use CQL query like:
`query=id=<id1> or id=<id2> or ....`
Please remember that url and response body have limitations, so request users in batches.